src/node_buffer.cc

```diff
@@ -19,7 +19,11 @@
   } while (0)
 
 #define ARGS_THIS(argT)                                                     \
-  Local<Object> obj = argT;                                                 \
+  Handle<Value> _obj = argT;                                                \
+  if (!HasInstance(_obj)) {                                                 \
+    return env->ThrowTypeError("must be called on Buffer instance");        \
```

Comment:
@Fishrock123 problem is that's a massive break. Don't even think nan can deal with it in a backwards compatible way. Since `class` won't be out until 2 (since we're increasing the major, i guess...). Are we going to require this be a 3 release?